rfcs/convergence/authoring-stories.md

```diff
@@ -261,17 +261,112 @@ export const parameters = { controls: { expanded: true } };
 
 ### 5. how to author e2e suites for those stories
 
-TBA
+> This section assumes that the storybook stories forms the core documentation of Fluent converged.
+> This may change in the future so the measures proposed in this section might no longer be relevant
+> if this assumption is no longer true.
 
-### 6. how to properly annotate stories with TS metadata to get the best DX possible
+`react-menu` already has entire test suites dedicated to testing against storybook. Since each PR build will
+deploy a storybook, the cypress testing solution for converged components will test against the PR storybook. This
+includes the constraint that until a component is added to `react-components` it cannot be tested in CI.
 
-TBA
+The e2e testing solution for `react-menu` can be used for any converged component using storybook, and is also used by
+`react-accordion` and `react-popover`. The purposes of E2E tests currently is to run internaction tests against the browser.
+Visual regression tests are handled by screener in `app-vrtests` and is out of the scope of this RFC.
+
+We propose to write an NX workspace generator to setup the e2e testing folder structure and dependencies for each package. The process is quite simple currently but still needs to be done manually.
+
+How to write E2E tests is out of the scope of the RFC. Testing is encouraged to use publicly viewable stories as much as possible. However there will be edge cases and scenarios that will be tested which require stories that are accessible to the E2E test runner but not viewable by public consumers. We provide a proposal to handle this in a later section in this RFC.
+
+### 6. dissecting big story files into smaller ones
+
+For easier maintainability, each story should be in its own file. Common utilities can be shared in their own modules.
+We need to make sure that no code used in stories accidentally leak into the exported package. To make this more explicit
+in the repo, all colocated stories and related code should be stored under a `/stories` folder in the package. Below is
+an example of the proposed file tree:
+
+```
+packages/react-menu/
+├─ stories/
+│  ├─ Menu/
+│    ├─ Default.stories.tsx
+│    ├─ Controlled.stories.tsx
+│  ├─ MenuList/
+│    ├─ Default.stories.tsx
+│    ├─ WithIcons.stories.tsx
+│  ├─ tmp-icons.tsx
+├─ src/
+│  ├─ index.js
+├─ .gitignore
+├─ package.json
+├─ README.md
+├─ Spec.md
+```
+
+The name of each `stories` file should be the name of the story in storybook, for easy searching in the IDE. Since we follow
+`Component Story Format (CSF)` a different folder should be used for each component in storybook.
+
+We should invest in creating lint rules and automation that will enforce this file structure and guarantee imports
+from storybook related code do not leak into the package.
 
 ### 7. how structure stories in the storybook nav tree
 
-TBA
+> This section assumes that the storybook stories forms the core documentation of Fluent converged.
```

Comment:
Well we no longer need any of the proposals in this PR if that's the case, because we would never be showing consumers strange internal stories to test edge cases

Reply:
not sure I follow